src/saml20_clj/sp/request.clj

```diff
@@ -152,54 +137,54 @@
+(defn- build-logout-obj
+  ^LogoutRequest [issuer user-email idp-url instant request-id]
   (assert (non-blank-string? idp-url) "idp-url is required")
-  (assert (non-blank-string? user-email) "user-email is required")
   (assert (non-blank-string? issuer) "issuer is required")
-  (assert (non-blank-string? relay-state) "relay-state is required")
-  (add-query-params idp-url {:SAMLRequest (encode-decode/str->deflate->base64
-                                           (coerce/->xml-string (make-logout-request-xml
-                                                                 :idp-url idp-url
-                                                                 :request-id request-id
-                                                                 :issuer issuer
-                                                                 :user-email user-email)))
-                             :RelayState relay-state}))
+  (assert (non-blank-string? user-email) "user-email is required")
+  (doto (.buildObject (LogoutRequestBuilder.)
+                      SAMLConstants/SAML20P_NS
+                      "LogoutRequest"
+                      "samlp")
+    (.setID request-id)
+    (.setIssueInstant instant)
+    (.setDestination idp-url)
+    (.setIssuer (doto (.buildObject (IssuerBuilder.)
+                                    SAMLConstants/SAML20_NS
+                                    "Issuer"
+                                    "saml")
+                  (.setValue issuer)))
+    (.setNameID (doto (.buildObject (NameIDBuilder.)
+                                    SAMLConstants/SAML20_NS
+                                    "NameID"
+                                    "saml")
+                  (.setValue user-email))))
+  )
 
 (defn idp-logout-redirect-response
   "Return Ring response for HTTP 302 redirect."
   ([issuer user-email idp-url relay-state]
    (idp-logout-redirect-response issuer user-email idp-url relay-state (random-request-id)))
   ([issuer user-email idp-url relay-state request-id]
-   (let [url (logout-redirect-location
-              :idp-url idp-url
-              :user-email user-email
-              :issuer issuer
-              :relay-state relay-state
-              :request-id request-id)]
-     {:status  302 ; found
-      :headers {"Location" url}
-      :body    ""})))
+   (idp-logout-redirect-response {:issuer issuer
+                                  :user-email user-email
+                                  :idp-url idp-url
+                                  :relay-state relay-state
+                                  :request-id request-id}))
+  ([{:keys [request-id instant idp-url issuer user-email credential relay-state sig-alg]
+     :or {instant (t/instant)
+          request-id (random-request-id)
+          sig-alg -sig-alg}}]
+   (let [logout-request (build-logout-obj issuer user-email idp-url instant request-id)]
+     (redirect-response (setup-message-context logout-request credential sig-alg idp-url) relay-state))))
```

Comment:
note: Usually we order the vararg function clauses from fewest args to most. Though I see how this is kind-of a 8-arity version of it. Up to you if you want to reorder it in this case.

Reply:
I thought it was easier to read with the 'actually do stuff' case at the bottom of the form.